server/base/src/main/java/org/apache/accumulo/server/AbstractServer.java

```diff
@@ -91,6 +92,20 @@ protected AbstractServer(ServerId.Type serverType, ConfigOpts opts,
     ClusterConfigParser.validateGroupNames(List.of(resourceGroup));
     SecurityUtil.serverLogin(siteConfig);
     context = serverContextFactory.apply(siteConfig);
+
+    final String upgradePrepNode = Constants.ZPREPARE_FOR_UPGRADE;
+    try {
+      if (context.getZooSession().asReader().exists(upgradePrepNode)) {
+        throw new IllegalStateException(
+            "Instance has been prepared for upgrade, no servers can be started."
+                + " To undo this state and abort upgrade preparations delete the zookeeper node: "
+                + upgradePrepNode + " for instance: " + context.getInstanceID());
```

Comment:
The ZK path in the error message will be relative because of chroot, so can not use the exact path in the error message to delete.  Should we prepend the root path for this case?

Reply:
Updated error message to include full path in 36e4fda